### PR TITLE
Add macOS aarch64 clock cycle count

### PR DESCRIPTION
**Pull request by ymlawk**
Add cycle count for macOS aarch64, which use Apple's interface.

### CODE REVIEW

File: src/macOS_aarch64_cpmu.h [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+*/
+
+#ifndef MACOS_AARCH64_CPMU_H_
+#define MACOS_AARCH64_CPMU_H_
+
+#include "internal_macros.h"
+
+#ifdef BENCHMARK_MACOS_AARCH64
+
+bool configure_macOS_rdtsc();
```
Comment (kriskwiatkowski) [inline src/macOS_aarch64_cpmu.h:22]: @ymlawk Can you add some basic comments to those functions (mainly meaning of return values).

File: src/macOS_aarch64_cpmu.cc [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  http://blog.stuffedcow.net/2013/05/measuring-rob-capacity/
+  2014-10-14
+*/
+
+#include "macOS_aarch64_cpmu.h"
+
+#ifdef BENCHMARK_MACOS_AARCH64
+
+#include <dlfcn.h>
+#include <pthread.h>
```
Comment (kriskwiatkowski) [inline src/macOS_aarch64_cpmu.cc]: Is it needed? note gbench uses pthreads only if BENCHMARK_HAS_PTHREAD_AFFINITY is defined. Can we do something similar here?

File: src/macOS_aarch64_cpmu.cc [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+  //  g_config[5] = CPMU_SYNC_BR_ANY_MISP | CFGWORD_EL0A64EN_MASK;
+
+  return configure_macOS_rdtsc();
+}
+
+unsigned long long int macOS_rdtsc() {
+  if (kpc_get_thread_counters(0, COUNTERS_COUNT, g_counters)) {
+    return 0;
+  }
+  return g_counters[0 + 2];
```
Comment (kriskwiatkowski) [inline src/macOS_aarch64_cpmu.cc:141]: 0 + 2?

File: src/macOS_aarch64_cpmu.cc (hunk 2 above)
Comment (ymlawk) [reply src/macOS_aarch64_cpmu.cc]: dlfcn.h is required for dlsym on line 99. pthread is removed as we don't need to set thread QoS like the reference.

File: src/macOS_aarch64_cpmu.cc (hunk 3 above)
Comment (ymlawk) [reply src/macOS_aarch64_cpmu.cc:141]: It it interesting that the config is for the configurable counters, which right after the first 2 constant counters. So, if you configure 0, means that the counter value is at 0 + 2. If you configure 3, means that the counter value is at 3 + 2. Writing in this way is to remind us to read the correct counter.

**Pull request merged**